src/legacyilibassemble.js

```diff
@@ -0,0 +1,256 @@
+#!/usr/bin/env node
+/*
+ * legacyilibassemble.js - assemble of legacy style of ilib files.
+ *
+ * Copyright © 2024 JEDLSoft
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import fs from 'fs';
+import path from 'path';
+import Locale from 'ilib-locale';
+import UglifyJS from 'uglify-js-export';
+
+let ilibPath;
+let incPath;
+let outDir;
+let locales;
+let outFileName;
+let isCompressed;
+let jsFileList = [];
+let allJSList = [];
+
+const reDependentPattern = new RegExp(/require\(\s*(\"|\')\.*\/(.*\.js)(\"|\')\)\;/, "g");
+const reDataPattern = new RegExp(/\/\/\s*!data\s*(([^\\])+)/, "g");
+
+function assembleilib(options) {
+    ilibPath = options.opt.ilibPath || ".";
+    incPath = options.opt.ilibincPath || "./src/ilib-assemble-inc.js";
+    locales = options.opt.locales;
+    outDir = options.args[0];
+    outFileName = options.opt.outjsFileName || "ilib-assemble.js";
```

Comment:
That is the same name as this package... maybe a different default output file name? like "ilib-assembled.js" or "ilib-code-and-data.js" or "ilib-all.js" or something like that to make it clear that it is assembled code with data.

Reply:
I update to have the different default name `ilib-all.js` and `ilib-all-inc.js`. so I also modified all the places where the related file names are written.